File: CSharp/Rectangle.cs
namespace Rectangles{

    class Rectangle
    {
        public double Width;
        public double Height;
        public double X;
        public double Y;

        public Rectangle()   //default (primary) constructor
        {
            Width = 1;
            Height = 1;
            X = 0;
            Y = 0;
        }

        public Rectangle(double width, double height, double x, double y)   //parameterized constructor
        {
            Width = width;
            Height = height;
            X = x;
            Y = y;
        }

        public Rectangle(Rectangle other)   //copy constructor
        {
            Width = other.Width;
            Height = other.Height;
            X = other.X;
            Y = other.Y;
        }
        

        public double Area()
        {
            return Width * Height;
        }

        public double Perimeter()
        {
            return 2 * (Width + Height);
        }

        public (double, double)[] getVertices()
        {
            return new (double, double)[]
            {
                (X, Y),                     //bottom left
                (X + Width, Y),             //bottom right
                (X + Width, Y + Height),    //top right
                (X, Y + Height)             //top left
            };
        }

        public (double, double) getData()
        {
            return (Width, Height);
        }

        ~Rectangle()
        {
            
        }
    }
}

File: CPlusPlus/Rectangle.cpp
#include "Rectangle.h"
#include <iostream>

Rectangle::Rectangle() : width(10), height(10), x(0), y(0) {}

Rectangle::Rectangle(double width, double height, double x, double y) {
    this->width = width;
    this->height = height;
    this->x = x;
    this->y = y;
}

Rectangle::Rectangle(const Rectangle& other) {}


Rectangle::~Rectangle() {}

double Rectangle::area() {
    return width * height;
}

double Rectangle::perimeter() {
    return 2 * (width + height);
}

double Rectangle::get_width() const {
    return width;
}

double Rectangle::get_height() const {
    return height;
}

double Rectangle::get_x() const {
    return x;
}

double Rectangle::get_y() const {
    return y;
}